core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +2000,1275 @@ def export_data(
+
+    @classmethod
+    def fetch_page(
+        cls,
+        page_size: int,
+        offset: int,
+        sort_by: Optional[str],
+        topic_ids: Optional[List[str]],
+        last_activity: Optional[int],
+        language_code: str
+    ) -> Tuple[Sequence[TranslationSubmitterTotalContributionStatsModel],
+                int,
+                bool]:
+        """Returns the models according to values specified.
+
+        Args:
+            page_size: int. Number of models to fetch.
+            offset: int. Number of results to skip from the beginning of all
+                results matching the query.
+            sort_by: str|None. A string indicating how to sort the result.
+            topic_ids: List[str]|None.
+                List of topics user has contributions.
+            last_activity: int. Number of days within which a user has
+                contributed.
+            language_code: str. Language Code for filter.
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models with supplied filters and sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+
+        sort_options_dict = {
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingLastActivity']:
+                -cls.last_contribution_date,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingLastActivity']:
+                cls.last_contribution_date,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingPerformance']:
+                cls.recent_performance,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingAccuracy']:
+                -cls.overall_accuracy,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingAccuracy']:
+                cls.overall_accuracy,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingSubmissions']:
+                -cls.submitted_translations_count,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingSubmissions']:
+                cls.submitted_translations_count
+        }
+
+        sort = -cls.recent_performance
+        if sort_by is not None:
+            sort = sort_options_dict[sort_by]
+
+        # The first sort property must be the same as the property to which
+        # an inequality filter is applied. Thus, the inequality filter on
+        # last_activity can not be used here. Learn more about this here
+        # https://cloud.google.com/appengine/docs/legacy/standard/go111/datastore/query-restrictions#properties_used_in_inequality_filters_must_be_sorted_first.
+        if topic_ids is not None:
+            sort_query = cls.query(
```

Comment:
You will need to exercise this manually on your local server to generate the index.yaml file indexes needed to serve this query. https://cloud.google.com/appengine/docs/legacy/standard/python/config/indexref#creating_index_files

Reply:
How can I run the queries manually on the local server. I am bit confused here. The documentation you linked suggests to run the queries which will create a index.yaml file automatically. Should I manually type-in the values?

Reply:
https://github.com/oppia/oppia/wiki/Debugging-datastore-locally might be useful.

Reply:
I am trying this might take some time, I figuring out something that will help me through out the project and not just for this case.

Reply:
Thanks. That helps.

Reply:
Hi @seanlip I tried running the queries manually but nothing was generated in index.yaml. I have added the indexes with all possible queries in our models. PTAL.

Reply:
That's a bit odd, usually we have the opposite problem (i.e. people seeing unexpected changes in index.yaml). Maybe that will trigger when you run the devserver once you've hooked everything up, but for now it's also fine to populate them manually. Thanks.

Reply:
Actually I tested it on the devserver and ran some queries, some indexes were generated for that matter but added the missed ones.